src/controllers/scripting/legacy/controllerscriptinterfacelegacy.cpp

```diff
@@ -136,6 +136,15 @@ QJSValue ControllerScriptInterfaceLegacy::getSetting(const QString& name) {
     }
 }
 
+QObject* ControllerScriptInterfaceLegacy::getPlayer(const QString& deck) {
+    auto* const player = m_pScriptEngineLegacy->getPlayer(deck);
+    auto* const jsEngine = qjsEngine(this);
+    auto stateEnum = jsEngine->newQMetaObject(
+            QMetaType::fromType<JavascriptPlayerProxy::State>().metaObject());
+    jsEngine->newQObject(player).setProperty("State", stateEnum);
+    return player;
```

Comment:
Should I return the result of `jsEngine->newQObject` here? I'm unsure but I suppose `Q_INVOKABLE` transforms the `QObject` to a `QJSValue` so since that's what I'm already doing here, better directly return it?